tests/test_flask_server.py

```diff
@@ -111,3 +112,119 @@ def test_report_one_job_as_failure(job_manager):
                          content_type='application/json')
     assert result.status_code == 200
     assert job_manager.num_unassigned() == 1
+
+
+class DummyJobManager:
```

Comment:
I am not sure why we would use DummyJobManager class, while we could use MockJobManager!

Reply:
My tests didn't need any functionality of the `JobManager`, and I wanted to try making a dummy object (one that doesn't do anything and just stands in for another object).  But I see that doing so ultimately adds complexity.  Further, when I replaced all instances of `DummyJobManager` with `MockJobManger`, all the tests still passed.  So, I went with your suggestion, and simpler code.

Good idea!